### PR TITLE
Now happening and several fixes

### PR DESCRIPTION
**Pull request by mic4ael**

### CODE REVIEW

File: indico/MaKaC/webinterface/wcomponents.py [hunk 1]
```diff
@@ -1799,80 +1800,50 @@ def __init__(self,conf, tz=None):
...
         # This will contain a string formated for use in the template
         # javascripts
-        nowHappeningArray = None
-
-        # currently happening:
-        n = nowutc()
-        entries = self._conf.getSchedule().getEntriesOnDate(n)
-        entryCaptions = []
+        now_happening = None
+        current_dt = nowutc()
```
Comment (ThiefMaster) [inline indico/MaKaC/webinterface/wcomponents.py]: `nowutc(exact=False)` to avoid having seconds in it

Comment (mic4ael) [inline indico/MaKaC/webinterface/wcomponents.py]: But, does it matter here?

File: indico/modules/events/__init__.py [hunk 2]
```diff
@@ -289,3 +289,10 @@ def _render_event_ical_export(event, **kwargs):
     return render_template('events/display/event_ical_export.html', item=event,
                            ics_url=url_for('events.export_event_ical', event),
                            **get_base_ical_parameters(session.user, event, 'events'))
+
+
+@template_hook('contribution-fields')
+def _render_contribution_fields(event, **kwargs):
+    from indico.modules.events.contributions.models.fields import ContributionField
+    return render_template('events/management/contribution_field_list.html', event=event,
+                           fields=event.contribution_fields.order_by(ContributionField.position))
```
Comment (ThiefMaster) [inline indico/modules/events/__init__.py]: are you sure the `order_by` here is needed? it's specified in the relationship

File: indico/modules/events/templates/display/now_happening.html [hunk 3]
```diff
@@ -0,0 +1,48 @@
+<div id="now-happening" {{ text_color }} style="margin: 10px 0;">
+    <span id="now-happening-label" style="font-weight: bold;">
+        {% trans %}Now happening:{% endtrans %}
+    </span>
+    <span id="now-happening-entries">
+        {% for entry in entries %}
+            {% set obj = entry.object %}
+            {% if entry.type == timetable_entry_type.CONTRIBUTION %}
```
Comment (ThiefMaster) [inline indico/modules/events/templates/display/now_happening.html]: `entry.type.name == 'CONTRIBUTION'` - no need to pass the whole enum object to the template that way

File: indico/modules/events/templates/display/now_happening.html [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                {% set url = url_for('contributions.display_contribution', obj) %}
+            {% elif entry.type == timetable_entry_type.BREAK %}
+                {% if entry.parent %}
+                    {% set url = url_for('sessions.display_session', entry.parent.object) %}
+                {% else %}
+                    {% set url = url_for('timetable.timetable', event) %}
+                {% endif %}
+            {% endif %}
+            <span class="now-happening-entry" style="display: none;">
+                <a href="{{ url }}">{{ obj.title }}{% if obj.room_name %}{{ obj.room_name }}{% endif %}</a>
```
Comment (ThiefMaster) [inline indico/modules/events/templates/display/now_happening.html]: no space or anything else between title and room name?!

Comment (mic4ael) [inline indico/modules/events/templates/display/now_happening.html]: That is what we used to show before

File: indico/modules/events/templates/display/now_happening.html [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+<script>
+    (function() {
+        'use strict';
+
+        var entries = $('#now-happening .now-happening-entry');
+        var index = 0;
+
+        function showNextEntry() {
+            if (index >= entries.length) {
```
Comment (ThiefMaster) [inline indico/modules/events/templates/display/now_happening.html]: `index = index % entries.length;`

File: indico/modules/events/templates/display/now_happening.html [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+                index = 0;
+            }
+
+            entries.eq(index - 1).fadeOut(100);
+            entries.eq(index++).fadeIn('slow');
```
Comment (ThiefMaster) [inline indico/modules/events/templates/display/now_happening.html]: I'd use `index` here and `index++;` in a new line

File: indico/modules/events/templates/display/now_happening.html [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        }
+
+        showNextEntry();
+        _.defer(function() {
```
Comment (ThiefMaster) [inline indico/modules/events/templates/display/now_happening.html]: not needed

File: indico/modules/events/templates/management/_contribution_field_list.html [hunk 8]
```diff
@@ -0,0 +1,14 @@
+{% macro render_contribution_field_list(fields) %}
+    <div id="contribution-field-list">
+        {% for field in fields %}
+            <div class="contribution-field">
+                <span class="field-title">{{ field.title }}</span>
+                {% if field.field_type %}
+                    <span class="field-type">({{ field.field_type|replace('_', ' ') }})</span>
```
Comment (ThiefMaster) [inline indico/modules/events/templates/management/_contribution_field_list.html]: :monkey:

`field.field.friendly_name`

File: indico/modules/events/templates/display/now_happening.html (hunk 4 above)
Comment (ThiefMaster) [inline indico/modules/events/templates/display/now_happening.html]: that sounds awful(ly broken)

Comment (ThiefMaster) [inline indico/modules/events/templates/display/now_happening.html]: `{% if obj.room_name %} ({{ obj.room_name }}){% endif %}`

File: indico/modules/events/templates/display/now_happening.html [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+            setInterval(showNextEntry, 5000);
+        });
+    })(window);
```
Comment (ThiefMaster) [inline indico/modules/events/templates/display/now_happening.html]: no need for the `window`

File: indico/modules/events/timetable/templates/timetable_pdf_export.html [hunk 10]
```diff
@@ -17,6 +17,21 @@
     {% endcall %}
     {% call form_footer(form) %}
         <input class="i-button big highlight" type="submit" value="{% trans %}Generate{% endtrans %}">
-        <button type="button" class="i-button big" data-button-back>{% trans %}Back{% endtrans %}</button>
+        <a href="{{ back_url }}" class="i-button big">{% trans %}Back{% endtrans %}</a>
     {% endcall %}
+    <script>
+        $('[name=document_settings][value=showCoverPage], [name=document_settings][value=showTableContents], ' +
```
Comment (ThiefMaster) [inline indico/modules/events/timetable/templates/timetable_pdf_export.html]: at least specify some parent selector (e.g. the form).. querying fields globally by nothing but their name like this is pretty ugly

File: indico/web/forms/fields.py [hunk 11]
```diff
@@ -420,6 +421,12 @@ def _get_person_link(self, data, extra_data=None):
         if not person_link:
             person_link = self.person_link_cls(person=person)
         person_link.populate_from_dict(person_data)
+        email_address = data.get('email', '')
+        if email_address != person_link.email:
+            if not self.event.persons.filter_by(email=email_address).first():
+                person_link.person.email = email_address
+            else:
+                raise UserValueError(_('There is already a person with email {}').format(email_address))
```
Comment (ThiefMaster) [inline indico/web/forms/fields.py]: //the email..

File: indico/web/forms/fields.py [hunk 12]
```diff
@@ -420,6 +421,12 @@ def _get_person_link(self, data, extra_data=None):
         if not person_link:
             person_link = self.person_link_cls(person=person)
         person_link.populate_from_dict(person_data)
+        email_address = data.get('email', '')
```
Comment (ThiefMaster) [inline indico/web/forms/fields.py]: i think usually we just call it `email`

File: indico/modules/events/templates/management/_contribution_field_list.html (hunk 8 above)
Comment (mic4ael) [inline indico/modules/events/templates/management/_contribution_field_list.html]: ah, didn't know about that

**Pull request merged**